OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -51,7 +51,7 @@ public FindResources(Actor self, CPos avoidCell)
 
 		public override Activity Tick(Actor self)
 		{
-			if (IsCanceled || NextActivity != null)
+			if (IsCanceled || NextInQueue != null)
 				return NextActivity;
```

Comment:
Yes. If it's canceled, `nextActivity` will be null, and we need to return to the parent.